### PR TITLE
Move useRefEffect hook to the react-hooks package

### PR DESCRIPTION
**Pull request by behowell**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Move the hook useRefEffect to the react-hooks package and add tests.

### CODE REVIEW

File: change/@fluentui-react-next-2020-07-15-21-26-39-useRefEffect.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "prerelease",
+  "comment": "Move useRefEffect to hook to the react-hooks package",
```
Comment (khmakoto) [inline change/@fluentui-react-next-2020-07-15-21-26-39-useRefEffect.json:3]:
```suggestion
  "comment": "Move useRefEffect hook to the react-hooks package.",
```

File: change/@uifabric-react-hooks-2020-07-15-21-26-39-useRefEffect.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "minor",
+  "comment": "Add useRefEffect to hook to the react-hooks package",
```
Comment (khmakoto) [inline change/@uifabric-react-hooks-2020-07-15-21-26-39-useRefEffect.json:3]:
```suggestion
  "comment": "Add useRefEffect hook to the react-hooks package.",
```

File: change/@fluentui-react-next-2020-07-15-21-26-39-useRefEffect.json (hunk 1 above)
Comment (behowell) [reply change/@fluentui-react-next-2020-07-15-21-26-39-useRefEffect.json]: Oops, thanks for catching the typo!

**Comment by ecraig12345** [comment]
Can you also add this to the package readme? Otherwise looks good.

File: packages/react-hooks/src/useRefEffect.test.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    wrapper.setProps({});
+
+    expect(ref.current).toBe(prevRef);
+    expect(log).not.toHaveBeenCalled();
+  });
+
+  it('clears ref.current and calls cleanup when the component is unmounted', () => {
+    const prevRef = ref.current;
+
+    wrapper.unmount();
```
Comment (xugao) [inline packages/react-hooks/src/useRefEffect.test.tsx:56]: hmm we shouldn't write individual test this way which depends on the previous test to run. i always expect i can run one test and it should be pass by itself. in your case it's better to combine these into one test. adding comments between the steps will help readability

**Comment by behowell** [comment]
I updated the API for `useRefEffect` to return a single ref callback function, which also has a `.current` member. This simplifies the API and reduces the ability to misuse it by setting an element's ref to the `RefObject` rather than the `RefCallback` (they are now one in the same).

**Pull request merged**